bin/agitator

```diff
@@ -157,10 +157,13 @@ function start_agitator() {
 
 function stop_agitator() {
   [[ -n $AGITATOR_USER ]] || AGITATOR_USER=$(whoami)
-  echo "Stopping all processes matching 'sleep' as $AGITATOR_USER"
-  pkill -f sleep 2>/dev/null
-  echo "Stopping all processes matching 'agitator' as $AGITATOR_USER"
-  pkill -f agitator 2>/dev/null
+  echo "Stopping all processes in the same process group as 'agitator' as $AGITATOR_USER"
+  ## get process ids of agitator processes then just keep the first
+  agitator_pid=$(pgrep -f agitator | head -1)
+  ## get the group process id of the agitator process then clear whitespace
+  group_pid=$(ps -p "${agitator_pid}" -o pgid= | xargs)
+  ## kill all processes in process group, should be all agitators and their sleep processes
+  kill -- -"${group_pid}"
```

Comment:
I think this will work better if you want to make use of arrays to get all agitator processes

```suggestion
  ## get process ids of all agitator processes (return 1 if none found)
  local agitator_pids=(); agitator_pids=($(pgrep -f agitator)) || return 1
  ## get the group process ids of all the agitator processes
  local group_pids=(); group_pids=($(ps -o pgid= -p "${agitator_pids[@]}"))
  ## kill all processes in the process groups (should include agitators and their sleep processes)
  kill -- "${group_pids[@]/#/-}"
```

Reply:
This looks like its working. I added some print statements to see what this was doing and everything checks out. The only thing I noticed was the `pgrep -f agitator` call is actually picking up the `pid` of the `./bin/agitator stop` bash script itself which ultimately is killed via its gpid further along in the script. Not sure if this could cause some issues or not. Maybe we could pgrep for the string "agitator start" to avoid this. I'll give that a try.

Reply:
Addresses in e768fc1